lib/chef-dk/command/generator_commands/cookbook.rb

```diff
@@ -195,9 +203,11 @@ def read_and_validate_params
           @cookbook_name_or_path = arguments[0]
           if !@cookbook_name_or_path
             @params_valid = false
-          elsif /-/ =~ File.basename(@cookbook_name_or_path)
-            err("Hyphens are not allowed in cookbook names. Please specify a cookbook name without hyphens.")
-            @params_valid = false
+          elsif /-/ =~ File.basename(@cookbook_name_or_path) && config[:skip_prompt] == false
+            warn("Hyphens are not allowed in cookbook names.")
```

Comment:
This is not correct, we should make the warning stern but accurate. How about "hyphens can cause problems in cookbooks names, and are not recommended"?